### PR TITLE
intergrate opensearch v2

### PR DESCRIPTION
**Pull request by wenchajun**
Signed-off-by: chengdehao <dehaocheng@kubesphere.io>

### What type of PR is this?

/kind feature

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/simple/client/es/client.go [hunk 1]
```diff
@@ -34,9 +36,10 @@ import (
 )
 
 const (
-	ElasticV5 = "5"
-	ElasticV6 = "6"
-	ElasticV7 = "7"
+	ElasticV5    = "5"
+	ElasticV6    = "6"
+	ElasticV7    = "7"
+	OpenSearchV2 = "2"
```
Comment (benjaminhuo) [inline pkg/simple/client/es/client.go:42]: The version can be changed like below to distinguish it with elasticsearch versions
```suggestion
	OpenSearchV2 = "OpenSearchV2"
```

Comment (wenchajun) [reply pkg/simple/client/es/client.go:41]: We determine the version by the `version` field in the data returned in the response, which, if set up this way, is not well handled on the back end.
https://github.com/kubesphere/kubesphere/blob/master/pkg/simple/client/es/client.go#L125

File: pkg/simple/client/es/versions/opensearch/v2/v2.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2020 KubeSphere Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
```
[the remark this answers is not in the document]
Comment (wenchajun) [reply pkg/simple/client/es/versions/opensearch/v2/v2.go]: fixed

File: pkg/simple/client/es/client.go (hunk 1 above)
Comment (benjaminhuo) [reply pkg/simple/client/es/client.go:41]: Is there some field returned in the response that contains `opensearch`?

Comment (wenchajun) [reply pkg/simple/client/es/client.go:41]: This field is returned, so this constant I think is better set to `2`

Comment (benjaminhuo) [reply pkg/simple/client/es/client.go:41]: I mean the distribution field contains "opensearch"
So if this field is "opensearch", we can add a `opensearchv` prefix to the version 2 and that will make the version become `opensearchv2`

Comment (wenchajun) [reply pkg/simple/client/es/client.go:41]: a5b793b04e2efd33257844887610184
I tested the return fields of elasticsearch and there is no `distribution` field in es, which should be the difference between elasticsearch and OpenSearch.

Comment (benjaminhuo) [reply pkg/simple/client/es/client.go:41]: So when there is distribution field and the value is opensearch, we can add `opensearchv` prefix to the v

**Pull request merged**